moveit_ros/visualization/motion_planning_rviz_plugin/src/motion_planning_frame_planning.cpp

```diff
@@ -108,7 +114,77 @@ void MotionPlanningFrame::onClearOctomapClicked()
   clear_octomap_service_client_.call(srv);
 }
 
-void MotionPlanningFrame::computePlanButtonClicked()
+void MotionPlanningFrame::computeCartecianPlanButtonClicked()
+{
+  if (!move_group_)
+    return;
+
+  // Clear status
+  ui_->result_label->setText("Planning...");
+
+  // get start and goal points
+  std::vector<geometry_msgs::Pose> waypoints;
+  robot_state::RobotState start = *planning_display_->getQueryStartState();
+  robot_state::RobotState goal = *planning_display_->getQueryGoalState();
+  geometry_msgs::Pose start_msg;
+  geometry_msgs::Pose goal_msg;
+  tf::poseEigenToMsg(start.getGlobalLinkTransform(move_group_->getEndEffectorLink()), start_msg);
+  tf::poseEigenToMsg(goal.getGlobalLinkTransform(move_group_->getEndEffectorLink()), goal_msg);
+  waypoints.push_back(start_msg);
+  waypoints.push_back(goal_msg);
+
+  // setup default params
+  double cart_step_size = 0.01;
+  double cart_jump_thresh = 0.0;
+  bool avoid_collisions = false;
+
+  // compute trajectory
+  current_plan_.reset();
+  moveit_msgs::RobotTrajectory trajectory;
+  double fraction =
+      move_group_->computeCartesianPath(waypoints, cart_step_size, cart_jump_thresh, trajectory, avoid_collisions);
+  Q_EMIT planningFinished();
+
+  if (fraction > 0)
+  {
+    // Success
+    ui_->result_label->setText(QString::number(fraction * 100, 'f', 0).append(" % achieved"));
+
+    moveit_msgs::RobotTrajectory scaled_trajectory = moveit_msgs::RobotTrajectory(trajectory);
+
+    // Scaling (https://groups.google.com/forum/#!topic/moveit-users/MOoFxy2exT4)
+
+    // The trajectory needs to be modified so it will include velocities as well.
+    // First to create a RobotTrajectory object
+    robot_trajectory::RobotTrajectory rt(move_group_->getRobotModel(), move_group_->getName());
+
+    // Second get a RobotTrajectory from trajectory
+    rt.setRobotTrajectoryMsg(*move_group_->getCurrentState(), scaled_trajectory);
+
+    // Thrid create a IterativeParabolicTimeParameterization object
+    trajectory_processing::IterativeParabolicTimeParameterization iptp;
+
+    // Fourth compute computeTimeStamps
+    bool success =
+        iptp.computeTimeStamps(rt, ui_->velocity_scaling_factor->value(), ui_->acceleration_scaling_factor->value());
```

Comment:
So this is the current workaround for velocity scaling cartesian paths? I learned something already.

Reply:
I don't know any, and I actually failed to implement it in my own code, but I can confirm that it works as you wrote it here :)